lib/AST/TypeRepr.cpp

```diff
@@ -147,7 +147,7 @@ SourceLoc TypeRepr::findAttrLoc(TypeAttrKind kind) const {
     for (auto attr : attrTypeRepr->getAttrs()) {
       if (auto typeAttr = attr.dyn_cast<TypeAttribute*>())
         if (typeAttr->getKind() == kind)
-          return typeAttr->getAttrLoc();
+          return typeAttr->getStartLoc();
```

Comment:
I was looking for what this change is going to affect and found https://github.com/apple/swift/blob/main/lib/Sema/TypeCheckDeclPrimary.cpp#L1803-L1806 which instead of `1` should also use `strlen("@retroactive ")`? We have a few tests for that diagnostic but none of them check fix-it :/

Reply:
Sure, I'll do that. I feel like there should be a better way than that `strlen`, but I wasn't up for adding more infrastructure right now.

Reply:
What devilry is this? The `.getAdvancedLoc(1)` thing is actually working for `@retroactive`, and I don't understand why

Reply:
That is weird...